### PR TITLE
Remove all deprecated lsp packages whose functionality is part of lsp

### PR DESCRIPTION
**Pull request by dmacvicar**
All the packages are now part of _lsp-mode_ itself. Most of them even mention it in the README.

Superseedes #5971 and #5894

Left are:

* lsp-hack
* lsp-intellij
* lsp-javacomp
* lsp-haskell
* lsp-java
* lsp-javascript-flow
* lsp-p4

### Brief summary of what the package does

Example:

```markdown
# DEPRECATED

You do not need this package anymore! Just use `lsp-mode` directly. Read [the docs](https://github.com/emacs-lsp/lsp-mode/#configuration).

tl;dr:
```

### Direct link to the package repository

https://github.com/emacs-lsp
* lsp-clangd
* lsp-css
* lsp-dart
* lsp-fortran
* lsp-go
* lsp-html
* lsp-javascript-typescript
* lsp-ocaml
* lsp-php
* lsp-python
* lsp-ruby
* lsp-rust
* lsp-sh
* lsp-typescript
* lsp-vue

### Your association with the package

User

### Relevant communications with the upstream package maintainer

//cc @gpittarelli

### Checklist

N/A

### CODE REVIEW

**Comment by tarsius** [comment]
@yyoncho has confirmed that it is time to merge this now, see https://github.com/melpa/melpa/pull/5894#issuecomment-456093040.
